### PR TITLE
Wq category no info

### PR DESCRIPTION
**Pull request by btovar**
Use largest resources seen in workers for categories that do not define a maximum allocation.

Experimental, do not merge yet.

### CODE REVIEW

**Comment by btovar** [comment]
@dthain: ready for review

File: work_queue/src/work_queue.h [hunk 1]
```diff
@@ -715,13 +732,20 @@ This function should only be called after @ref work_queue_empty returns true.
 */
 void work_queue_delete(struct work_queue *q);
 
-/** Add a log file that records the states of the connected workers and submitted tasks.
+/** Add a log file that records cummulative statistics of the connected workers and submitted tasks.
```
Comment (dthain) [inline work_queue/src/work_queue.h]: sp

File: work_queue/src/work_queue.c [hunk 2]
```diff
@@ -5933,8 +5871,147 @@ int work_queue_specify_log(struct work_queue *q, const char *logfile)
...
+	buffer_printf(&B, "CATEGORY %s FIRST %s ", c->name, mode);
+	rmsummary_print_buffer(&B, category_dynamic_task_max_resources(c, NULL, CATEGORY_ALLOCATION_FIRST), 1);
+	write_transaction(q, buffer_tostring(&B));
+	buffer_rewind(&B, 0);
+
+	buffer_free(&B);
+}
+
+
+int work_queue_specify_transactions_log(struct work_queue *q, const char *logfile) {
```
Comment (dthain) [inline work_queue/src/work_queue.c]: A self-documenting file, I like it!

File: work_queue/src/work_queue.c [hunk 3]
```diff
@@ -254,6 +258,7 @@ static void push_task_to_ready_list( struct work_queue *q, struct work_queue_tas
 static work_queue_task_state_t change_task_state( struct work_queue *q, struct work_queue_task *t, work_queue_task_state_t new_state);
 
 const char *task_state_str(work_queue_task_state_t state);
```
Comment (dthain) [inline work_queue/src/work_queue.c]: Should either be `static` or `work_queue_task_state_str` to fit the namespace.

**Comment by dthain** [comment]
As we discussed today, there is a lot going on in this pull request, please split up by purpose.
But I think the general approach is good.

**Comment by btovar** [comment]
Seuperseded by #1304, #1305, and #1306.

**Pull request closed without merging**